extendeddatasquare_test.go

```diff
@@ -96,6 +96,49 @@ func TestMarshalJSON(t *testing.T) {
 	}
 }
 
+func TestNewExtendedDataSquare(t *testing.T) {
```

Comment:
[note for reviewers] I tend to prefer table driven tests but I found it difficult to write a singular table driven test for all the test cases I wanted to cover the errors are different. 

If reviewers don't like this style of tests, I can try to refactor to something else.